google/cloud/storage/internal/object_requests.h

```diff
@@ -212,6 +212,40 @@ class UpdateObjectRequest
 
 std::ostream& operator<<(std::ostream& os, UpdateObjectRequest const& r);
 
+/**
+ * Represents a request to the `Objects: compose` API.
+ */
+class ComposeObjectRequest
+    : public GenericObjectRequest<
+          ComposeObjectRequest, EncryptionKey, Generation,
+          IfGenerationMatch, IfMetagenerationMatch, PredefinedAcl,
+          SourceEncryptionKey, UserProject> {
```

Comment:
I think you need to create a new class `DestinationPredefinedAcl` because `PredefinedAcl` will create the wrong query parameter, right?

Reply:
Whoops, Right!